Aqovia.Utilities.SagaMachine.Tests/StateMachineTests.cs

```diff
@@ -338,5 +356,38 @@ public void SagaMachineShouldHaveInitialiseStateId()
             _keyValueStoreMock.Verify(o => o.Remove(It.IsAny<string>(), It.IsAny<string>()), Times.Once);
         }
 
+        [Fact]
+        public void SagaMachine_Should_Not_Stop_Saga_If_An_Exception_Happens()
+        {
+            //Arrange
+            _keyValueStoreMock.Setup(o => o.GetValue<TestState>(It.IsAny<string>()))
+                .Returns(new HashedValue<TestState>
+                {
+                    Hash = Guid.Empty.ToString(),
+                    Value = new TestState()
+                });
+
+            // We need the below to set the return value
+            _keyValueStoreMock.Setup(kv => kv.Remove(It.IsAny<string>(), It.IsAny<string>())).Returns(true);
+
+            var messagePublisher = new MessagePublisherFake();
+
+            //Act
+            _sagaMachine = new SagaMachine<TestState>(_keyValueStoreMock.Object, messagePublisher.PublishMessage, _mockEventloggerFactory.Object);
+            _sagaMachine
+               .WithMessage<HelloMessage>((proccess, msg) => proccess
+                   .Publish((msgForPub, state) => new[] { new GoodbyeMessage() })
+                   .Publish((msgForPub, state) => new[] { new SomeOtherMessage() })
```

Comment:
A bit of a spooky action at distance, to trigger the throwing [in there](https://github.com/Aqovia/Aqovia.SagaMachine/pull/1/commits/c4b4972e66bf6fc0574e7dd35c82f090492afcb0#diff-45b6849196e8dd5f2d7b1a61cff96a2aR27) only when encountering the message of type. 
Why not make it simple and just throw on first message in there? Or does the test actually care about sending 'some messages' successfully first?